bigquery/vacancy-feedback-metrics-by-month.sql

```diff
@@ -1,83 +1,115 @@
+    SUM(hires_rate_through_tv*vacancies_published) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)/SUM(vacancies_published) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS hires_rate_through_tv_this_AY_so_far,
+    SUM(exclusive_hires_rate*vacancies_published) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)/SUM(vacancies_published) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS exclusive_hires_rate_this_AY_so_far,
+    SUM(exclusivity_rate*vacancies_published) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)/SUM(vacancies_published) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS exclusivity_rate_this_AY_so_far,
+    SAFE_DIVIDE(SUM(hires_through_tv) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW),
+      SUM(feedback_available) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)) AS raw_hires_rate_through_tv_this_AY_so_far,
+    SAFE_DIVIDE(SUM(exclusive_hires) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW),
+      SUM(feedback_available) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)) AS raw_exclusive_hires_rate_this_AY_so_far,
+    SAFE_DIVIDE(SUM(exclusive_listings) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW),
+      SUM(feedback_available) OVER (PARTITION BY AY_beginning ORDER BY month ASC ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)) AS raw_exclusivity_rate_this_AY_so_far,
   FROM (
     SELECT
-      DATE_TRUNC(publish_on,MONTH) AS month,
-      category,
-      COUNT(*) AS vacancies_published,
-      COUNTIF(hired_status IS NOT NULL
-        AND listed_elsewhere IS NOT NULL) AS feedback_available,
-      COUNTIF(hired_status="hired_tvs") AS hires_through_tv,
-      COUNTIF(hired_status IN ("hired_dont_know",
-          "hired_tvs")) AS hires_through_tv_upperbound,
-      COUNTIF(hired_status IN ("hired_tvs")
-        AND listed_elsewhere IN ("not_listed",
-          "listed_free")) AS exclusive_hires,
-      COUNTIF(hired_status IN ("hired_dont_know",
-          "hired_tvs")
-        AND listed_elsewhere IN ("not_listed",
-          "listed_free",
-          "listed_dont_know")) AS exclusive_hires_upperbound,
-      COUNTIF(listed_elsewhere IN ("not_listed","listed_free")) AS exclusive_listings
-    FROM
-      `teacher-vacancy-service.production_dataset.vacancies_published`
+      month,
+      AY_beginning,
+      SUM(vacancies_published) AS vacancies_published,
+      SUM(feedback_available) AS feedback_available,
```

Comment:
In SUM(feedback_available) does feedback_available refer directly to a column or something? I would be tempted to say that it's confusing to reassign feedback_available to something other than the column (or whatever) it originally referred to, because that namespace should be reserved for the column called feedback_available.

Reply:
(Same for vacancies_published)

Reply:
@david-mears-dfe it's taking a table of 'month-categories' with a number of vacancies/feedbacks for each one and collapsing it into a table of months, again with a number of vacancies/feedbacks for each one. I think I'll add in the _this_month prefix for clarity.